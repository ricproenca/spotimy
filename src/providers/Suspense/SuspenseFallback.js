import LinearProgress from '@mui/material/LinearProgress';
import React from 'react';

/**
 * Help with handling async operations by waiting for some code to load
 * and show a progress bar while waiting.
 */
const SuspenseFallback = () => <LinearProgress color='primary' data-testid='suspense-progress' />;

export default SuspenseFallback;
