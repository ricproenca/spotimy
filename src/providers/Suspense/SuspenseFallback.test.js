import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';

import SuspenseFallback from './SuspenseFallback';

describe('SuspenseFallback', () => {
  describe('Testing rendering', () => {
    test('It matches snapshot', () => {
      // Act
      const { asFragment } = render(<SuspenseFallback />);

      // Assert
      expect(asFragment()).toMatchSnapshot();
    });

    test('It renders', () => {
      // Act
      render(<SuspenseFallback />);
      const progressBar = screen.getByTestId('suspense-progress');

      // Assert
      expect(progressBar).toBeInTheDocument();
    });
  });
});
