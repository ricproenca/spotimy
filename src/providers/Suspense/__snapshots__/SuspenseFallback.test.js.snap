// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuspenseFallback Testing rendering It matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
    data-testid="suspense-progress"
    role="progressbar"
  >
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
    />
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
    />
  </span>
</DocumentFragment>
`;
